### DISCUSSION TITLE
Interpolating the plot

### DISCUSSION THREAD

**Discussion by bhawesh-bks** [pyvista/pyvista#3770] [Q&A]
I have a dataset comprising of three columns x,y and predicted lithology. I want to plot a map with continous boundaries but since i have very few values the map that i plot is discrete.
I have hereby attacking the dataset that can be used to plot the map and also a reference map.
[Book1.xlsx](https://github.com/pyvista/pyvista/files/10332191/Book1.xlsx)

![Screenshot (7)](https://user-images.githubusercontent.com/97894832/210246862-adb36ecb-6fc5-469b-8620-94a250e3539b.png)

**Comment by bhawesh-bks** [comment]
@banesullivan please sugggest an answer.

**Reply by banesullivan** [reply]
Well, these data are on a 2D plane so I don't think they'll look too interesting in 3D with PyVista, but here is a start:

I noticed after looking at the XY values that they are in a regular lattice, so I used `np.unique()` to find the unique nodes of that lattice to build a `pyvista.StructuredGrid`.

```py
import pyvista as pv
import numpy as np
import pandas as pd

df = pd.read_excel('/Users/bane.sullivan/Desktop/Book1.xlsx')
```

```py
>>> len(np.unique(df['x'])), len(np.unique(df['y']))
(93, 58)
```

```py
xy = df[['x', 'y']].values

points = np.c_[xy, np.zeros(len(xy))]

grid = pv.StructuredGrid()
grid.points = points
grid.dimensions = len(np.unique(df['x'])), len(np.unique(df['y'])), 1

grid['RF_pred'] = df['RF_pred'].values

p = pv.Plotter(notebook=0)
p.add_mesh(grid, n_colors=6)
p.show(cpos='xy')
```

I chose 6 for `n_colors` because that was the number of unique categories in `RF_pred`:

```py
>>> len(np.unique(df['RF_pred']))
6
```
![Screen Shot 2023-01-05 at 9 03 18 PM](https://user-images.githubusercontent.com/22067021/210927757-97f94665-28d9-4219-934f-4efc81476320.png)


Maybe we can use contours to improve it?

```py
cntrs, edges = grid.extract_surface().contour_banded(6)

p = pv.Plotter(notebook=0)
p.add_mesh(cntrs, n_colors=6)
p.add_mesh(edges, color='k')
p.show(cpos='xy')
```

--------

Honestly, for 2D data like this, I would just stick with Matplotlib:

```py
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
df = pd.read_excel('/Users/bane.sullivan/Desktop/Book1.xlsx')

x = df['x'].values
y = df['y'].values

nx = len(np.unique(x))
ny = len(np.unique(y))

RF_pred = df['RF_pred'].values.reshape(
    (nx, ny), 
    order='F'
).swapaxes(0, 1)

xx, yy = np.meshgrid(np.unique(x), np.unique(y))

plt.contourf(xx, yy, RF_pred)
```
![download](https://user-images.githubusercontent.com/22067021/210928589-d2c385d3-76a4-4ebe-90c9-46bd5dd56885.png)
